addons/website_sale/models/website.py

```diff
@@ -410,12 +410,13 @@ def _prepare_sale_order_values(self, partner_sudo):
         self.ensure_one()
         addr = partner_sudo.address_get(['delivery'])
         if not request.website.is_public_user():
-            # FIXME VFE why not use last_website_so_id field ?
             last_sale_order = self.env['sale.order'].sudo().search(
                 [('partner_id', '=', partner_sudo.id)],
```

Comment:
```suggestion
            last_sale_order = self.env['sale.order'].sudo().search(
                [('partner_id', '=', partner_sudo.id), ('website_id', '=', self.id)],
```
Wouldn't this be a good enough solution ?  That way, non ecommerce orders won't impact the default cart creation.
I prefer to avoid another magic fallback, especially for an unusual flow where a salesman/backend user has to manually input a wrong address that doesn't belong to the customer.

Reply:
@Feyensv ok sounds good, seems easier if we don't want to use addresses from non ecommerce SOs